src/plcc/parse_spec/parse_syntactic_spec/parse_syntactic_spec.py

```diff
@@ -0,0 +1,133 @@
+from typing import List
+from plcc.load_spec.load_rough_spec.parse_dividers import Divider
+from plcc.load_spec.load_rough_spec.parse_lines import Line
+import re
+from re import Match
+from .parse_syntactic_structs import (
+    SyntacticSpec,
+    SyntacticRule,
+    Symbol,
+    CapturingTerminal,
+    RepeatingSyntacticRule,
+    StandardSyntacticRule,
+    LhsNonTerminal,
+    RhsNonTerminal,
+    Terminal,
+    CapturingSymbol,
+    MalformedLHSError,
+    MalformedBNFError,
+)
+
+
+def parse_left(line: Line, lhs: str) -> LhsNonTerminal:
+    match = re.match(r"<(?P<non_terminal>\S*)>(?::(?P<alt_name>\S+))?\s*", lhs)
+    if match is None:
+        raise MalformedLHSError(line)
+
+    non_terminal = match["non_terminal"]
+    alt_name = match["alt_name"]
+    return LhsNonTerminal(non_terminal, alt_name)
+
+
+def parse_right(line: Line, rhs: str) -> List[Symbol]:
+    total = []
+    if not rhs:
+        return total
+    symbols = rhs.split(" ")
+    for symbol in symbols:
+        if symbol == "":
+            continue
+        if symbol.startswith("#") or symbol is None:
+            return total
+        total.append(parse_symbol(symbol))
+    return total
```

Comment:
This method is getting long, refactoring it would be helpful, I believe that this can be reduced to a couple lines while still handling all the logic.